### PR TITLE
Preparation for 4.4 RC.1 release

### PR DESCRIPTION
**Pull request by Konamiman**
### All Submissions:

* [ ] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [ ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

* Add changelog entries for the pull requests cherrypicked after 4.4 beta 1 as appropriate.
* Bump version name in `woocommerce.php` to `4.5.0-dev`.

### CODE REVIEW

**Review by vedanshujain — CHANGES_REQUESTED** [review]
Some minor changes needed, other than that looks good

File: CHANGELOG.txt [hunk 1]
```diff
@@ -109,6 +117,16 @@
 * Performance - Lazy Loading Atomic Components. #2777
 * Pefactor - Remove dashicon classes. #2848
 
+**WooCommerce Blocks 3.1.0**
```
[the remark this answers is not in the document]
Comment (Konamiman) [reply CHANGELOG.txt:120]: ...and missing in `readme.txt`. Wrong copypaste is it.

**Review by vedanshujain — APPROVED** [review]
Looks great!

File: CHANGELOG.txt [hunk 2]
```diff
@@ -109,6 +116,11 @@
...
+* Fix - Missing permissions_callback arg in StoreApi route definitions. #2926
+* Fix - 'Product Summary' in All Products block is not pulling in the short description of the product. #2913
+* Dev - Add query filter when searching for a table. #2886
+
```
Comment (peterfabian) [inline CHANGELOG.txt:123]: Can you please include changes from 4.3.1? I've just created PR also for default branch.

https://github.com/woocommerce/woocommerce/pull/27216

**Review by vedanshujain — APPROVED** [review]
LGTM!

**Review by peterfabian — APPROVED** [review]
LGTM now, thanks!

**Pull request merged**